### PR TITLE
feat: port websocket tests to node test runner

### PR DESCRIPTION
**Pull request by ilteoood**
## This relates to...

#2267

## Rationale

Port websocket tests to node test runner

## Changes

Remove tap and use node test runner for the websocket suite

### Features

N/A

### Bug Fixes

N/A

### Breaking Changes and Deprecations

None

## Status

- [X] I have read and agreed to the [Developer's Certificate of Origin][cert]
- [X] Tested
- [S] Benchmarked (**optional**)
- [S] Documented
- [X] Review ready
- [ ] In review
- [ ] Merge ready

[cert]: https://github.com/nodejs/undici/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: test/websocket/opening-handshake.js [hunk 1]
```diff
@@ -1,215 +1,217 @@
...
-  t.plan(2)
+  return new Promise((resolve, reject) => {
+    const server = createServer((req, res) => {
+      const key = req.headers['sec-websocket-key']
+      assert.ok(key)
 
-  const server = createServer((req, res) => {
-    const key = req.headers['sec-websocket-key']
-    t.ok(key)
+      const accept = createHash('sha1').update(key + uid).digest('base64')
```
[the remark this answers is not in the document]
Comment (ilteoood) [reply test/websocket/opening-handshake.js:196]: These lines were already there

**Review by mcollina — APPROVED** [review]
lgtm

**Review by KhafraDev — CHANGES_REQUESTED** [review]
seems to have issues on windows

**Comment by ilteoood** [comment]
@KhafraDev I've re-aligned everything with the main branch and tried to run the CI on my fork without changing anything: https://github.com/ilteoood/undici/actions/runs/7356075163/job/20025666784

It seems to be working

**Comment by KhafraDev** [comment]
if it started working without change, it might be flaky. Not something I'd want to merge unless it was one-off.

**Comment by ilteoood** [comment]
> if it started working without change, it might be flaky. Not something I'd want to merge unless it was one-off.

I've re-aligned with main before it started working, maybe is related to that. I have retried multiple times to retrigger it without luck

**Comment by mcollina** [comment]
@KhafraDev you ok for this landing?

**Pull request merged**